integration-tests/tests/integration_test.rs

```diff
@@ -5287,6 +5287,64 @@ fn test_error_generated_for_pod_with_nontrivial_move_constructor() {
     run_test_expect_fail("", hdr, rs, &["take_a"], &["A"]);
 }
 
+#[test]
+fn test_double_destruction() {
+    let hdr = indoc! {"
+        #include <stdio.h>
+        #include <stdlib.h>
+        // A simple type to let Rust verify the destructor is run.
+        struct DestructorFlag {
+            DestructorFlag() = default;
+            DestructorFlag(const DestructorFlag&) = default;
+            DestructorFlag(DestructorFlag&&) = default;
+
+            ~DestructorFlag() {
+                if (!flag) return;
+                if (*flag) {
+                    fprintf(stderr, \"DestructorFlag is already set\\n\");
+                    abort();
+                }
+                *flag = true;
+                // Note we deliberately do NOT clear the value of `flag`, to catch Rust calling
+                // this destructor twice.
+            }
+
+            bool *flag = nullptr;
```

Comment:
You could remove all of this (and just call it `NotTriviallyDestructible`), and everything in `rs` below. Just `~DestructorFlag() {}` makes it nontrivial, everything else is extra lines that aren't supposed to compile.

Reply:
Done, thanks